### PR TITLE
Return BatchDisposition Errors with LockTokenIDs

### PR DESCRIPTION
**Pull request by vsaroopchand**
Proposal for #128

### CODE REVIEW

File: batch_disposition.go [hunk 1]
```diff
@@ -16,6 +16,11 @@ type (
 		Status       MessageStatus
 		cursor       int
 	}
+
+	// BatchDispositionErrors returns LockTokenIDs with associated error.
+	BatchDispositionErrors struct {
+		Errors map[*uuid.UUID] error
```
Comment (devigned) [inline batch_disposition.go]: What do you think about making this implement error? I think it would be nice if this was just a specialized error.

Comment (vsaroopchand) [reply batch_disposition.go]: Good idea 👍 ; I'll fix it.

File: batch_disposition.go [hunk 2]
```diff
@@ -39,31 +44,35 @@ func (bdi *BatchDispositionIterator) Next() (uuid *uuid.UUID) {
...
 	}
-	return nil
+	return &BatchDispositionErrors {
+		Errors: errors,
+	}
 }
 
 // SendBatchDisposition updates the LockToken id to the desired status.
-func (q *Queue) SendBatchDisposition(ctx context.Context, iterator BatchDispositionIterator) error {
+func (q *Queue) SendBatchDisposition(ctx context.Context, iterator BatchDispositionIterator) *BatchDispositionErrors {
```
Comment (devigned) [inline batch_disposition.go]: I'm teetering back and for on returning an error and specializing the error as described above. Thoughts?

Comment (vsaroopchand) [reply batch_disposition.go]: I agree, touch call. I do like the custom error type (see latest changeset) - but also seems odd. Maybe OnError, OnSuccess handlers?

**Pull request merged**